src/coreclr/tools/superpmi/superpmi-shim-collector/icorjitcompiler.cpp

```diff
@@ -36,6 +36,23 @@ CorJitResult interceptor_ICJC::compileMethod(ICorJitInfo*                comp,
                                              uint32_t* nativeSizeOfCode            /* OUT */
                                              )
 {
+    // See if we are filtering the collection (currently by simple substring match)
+    //
+    char* filter = GetEnvironmentVariableWithDefaultA("SuperPMIShimFilter", "");
```

Comment:
Doesn't this leak?

Also seems like it might be expensive to query this on every compilation.

Reply:
Yeah, I can make it cheaper. Also I messed things up a bit when "polishing" for the PR. Let me revise.